recipe/build.sh

```diff
@@ -1,39 +1,47 @@
 #!/bin/bash
 
-osname=`uname`
-if [ $osname == Linux ]; then
-    export CC="gcc"
-    export CXX="g++"
-elif [ $osname == Darwin ]; then
-    export CC="clang"
-    export CXX="clang++"
+set -ex
+
+export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$BUILD_PREFIX/lib/pkgconfig
+export PKG_CONFIG=$BUILD_PREFIX/bin/pkg-config
+
+if [[ $CONDA_BUILD_CROSS_COMPILATION == "1" ]]; then
+  if [[ "${CMAKE_CROSSCOMPILING_EMULATOR:-}" == "" ]]; then
+    # Mostly taken from https://github.com/conda-forge/pocl-feedstock/blob/b88046a851a95ab3c676c0b7815da8224bd66a09/recipe/build.sh#L52
+    rm $PREFIX/bin/llvm-config
+    cp $BUILD_PREFIX/bin/llvm-config $PREFIX/bin/llvm-config
+    export LLVM_CONFIG=${PREFIX}/bin/llvm-config
+  else
+    # https://github.com/mesonbuild/meson/issues/4254
+    export LLVM_CONFIG=${BUILD_PREFIX}/bin/llvm-config
+  fi
 fi
 
-export PKG_CONFIG_PATH="${PREFIX}/lib/pkgconfig/:"${PKG_CONFIG_PATH}
-export CFLAGS="-I${PREFIX}/include "${CFLAGS}
-export LDFLAGS="-L${PREFIX}/lib "${LDFLAGS}
+meson setup builddir/ \
+  ${MESON_ARGS} \
+  --prefix=$PREFIX \
+  -Dplatforms=x11 \
+  -Dgles1=disabled \
+  -Dgles2=disabled \
+  -Dgallium-va=disabled \
+  -Dgbm=disabled \
+  -Dgallium-vdpau=disabled \
+  -Dshared-glapi=enabled \
+  -Dgallium-drivers=softpipe,llvmpipe \
+  -Degl=disabled \
+  -Dglx=disabled \
+  -Dllvm=enabled \
+  -Dshared-llvm=enabled \
+  -Dlibdir=lib \
+  -Dosmesa=true \
+  -Dvulkan-drivers=swrast \
+  -Dopengl=true \
+  -Dglx-direct=false \
+  || { cat builddir/meson-logs/meson-log.txt; exit 1; }
```

Comment:
Many of these disabled libraries are things we want our libmesa to provide for us. Conda-forge specifically omits many of the rendering libraries in their recipe stating that there are difficulties building them. I am assuming there are driver packages that they, and likely us as well, are missing in order to build the `mesa-libgl` or `mesa-libglx` libraries.

If these prove too difficult to build in our CI as well, we'll need to follow the same path conda-forge did with [glvnd](https://github.com/conda-forge/libglvnd-feedstock/blob/main/recipe/meta.yaml).

Reply:
I'd try with these on and start weeding them out as requirements are determined to be unavailable:
```suggestion
meson setup builddir/ \
  ${MESON_ARGS} \
  --prefix=$PREFIX \
  -Dlibdir=lib \
  -Dplatforms=x11 \
  -Dvulkan-drivers=all \
  -Dgallium-drivers=all \
  -Dgallium-va=disabled \
  -Dgallium-vdpau=disabled \
  -Dgles1=disabled \
  -Dgles2=disabled \
  -Dgbm=enabled \
  -Degl=enabled \
  -Dglvnd=enabled \
  -Dllvm=enabled \
  -Dshared-llvm=enabled \
  -Dlibunwind=enabled \
  -Dzstd=enabled \
  -Dglx-direct=false \
  || { cat builddir/meson-logs/meson-log.txt; exit 1; }
```

Reply:
I've split the recipe as a multi-output

Reply:
Is it not building `mesa-libglx`? I'm not seeing that as an output. I think we'd want it, though. Looks like Qt and things require it.

Reply:
Yes, `glx` is required by `qtbase` in the tests https://github.com/AnacondaRecipes/qtbase-feedstock/pull/7/files#diff-d1db07f6b4623b08d422eb0f882629e0742903fcd92b4968a45d5bb025315272R11-R45 but I wasn't able to resolve issues with it on linux.